packages/adblocker/src/engine/domains.ts

```diff
@@ -32,8 +36,21 @@ export class Domains {
     const notEntities: number[] = [];
     const hostnames: number[] = [];
     const notHostnames: number[] = [];
+    const rawParts: string[] = [];
 
     for (let hostname of parts) {
+      if (negate) {
```

Comment:
Nit. I think we could handle that below with `negation`, `start`, `end` instead of doing extra string slicing/interpolations but given there won't be a lot of these filters I think it's also fine as is.

Reply:
optimized in [`7b868fe` (#4524)](https://github.com/ghostery/adblocker/pull/4524/commits/7b868fe28b7eeaffb184a685aa28ebb3f49c6d04)